gateway-messages/src/sp_impl.rs

```diff
@@ -358,6 +360,31 @@ pub trait SpHandler {
         &mut self,
         key: [u8; 4],
     ) -> Result<&'static [u8], SpError>;
+
+    fn reset_component_prepare(
+        &mut self,
+        sender: SocketAddrV6,
+        port: SpPort,
+        component: SpComponent,
+    ) -> Result<(), SpError>;
+
+    // On success, this method will return unless the reset
+    // affects the SP_ITSELF.
```

Comment:
This is confusing to me, because we already have `reset_prepare` and `reset_trigger`.  Are there cases where we'd use `reset_component_trigger(... SpComponent::SP_ITSELF)` instead of `reset_trigger`?

Adding a comment to that effect would be helpful.

Reply:
After discussing with @jgallagher I removed `reset_prepare` and `reset_trigger` altogether.

Reply:
I realize I missed a few spots. I'm cleaning it up.